services/migrations/main_test.go

```diff
@@ -28,6 +30,11 @@ func assertTimeEqual(t *testing.T, expected, actual time.Time) {
 	assert.Equal(t, expected.UTC(), actual.UTC())
 }
 
+func assertTimeEqualEpsilon(t *testing.T, expected, actual time.Time) {
+	d := expected.UTC().Sub(actual.UTC()).Seconds()
```

Comment:
```suggestion
	d := expected.Sub(actual).Seconds()
```


`Time.UTC()` only changes the timezone in my mind, `Time.Seconds()` is correct and enough.

Reply:
I would just convert to unix timestamp and compare against - that way you have the second diff if needed + the exact time in log if it fails

Reply:
something like assert.Equal(t, expected.Unix(), actual.Unix(), "time differs") ...

Reply:
@6543 

> something like assert.Equal(t, expected.Unix(), actual.Unix(), "time differs") ...

It's impossible without a mocked time provider. The `expected = time.Now()` is not the same as the `createdAt = time.Now()` inside `services/migrations/gitlab.go`

Reply:
Aha, there is a `assert.WithinDuration()`, it's very handy. We do not need to re-invent the wheel `assertTimeEqualEpsilon`

Reply:
oh that's a new function! got added unoticed to me

Reply:
Good find! I completely missed the `assert.WithinDuration` method. It is used in other other place in the Gitea tests with a delta of 10 seconds and I reused this. It should be plenty even for slow machines.

---
[source](https://lab.fedeproxy.eu/forgefriends/forgefriends/-/merge_requests/37#note_915)